gax/src/streamingCalls/streaming.ts

```diff
@@ -158,21 +352,168 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
+      // added. see: https://github.com/googleapis/gax-nodejs/issues/65
+      stream.emit('response', {
+        code: 200,
+        details: '',
+        message: 'OK',
+        metadata,
+      });
+      this._responseHasSent = true;
+    });
+
+    stream.on('error', error => {
+      const timeout = retry.backoffSettings.totalTimeoutMillis;
+      const maxRetries = retry.backoffSettings.maxRetries!;
+      if ((maxRetries && maxRetries > 0) || (timeout && timeout > 0)) {
+        const e = GoogleError.parseGRPCStatusDetails(error);
+        let shouldRetry = this.defaultShouldRetry(e!, retry);
+        if (typeof retry.retryCodesOrShouldRetryFn! === 'function') {
+          shouldRetry = retry.retryCodesOrShouldRetryFn!(e!);
+        }
+
+        if (shouldRetry) {
+          if (maxRetries && timeout!) {
+            const newError = new GoogleError(
+              'Cannot set both totalTimeoutMillis and maxRetries ' +
+                'in backoffSettings.'
+            );
+            newError.code = Status.INVALID_ARGUMENT;
+            this.emit('error', newError);
+            this.destroy(newError);
+            return;
+          } else {
+            retryStream = this.retry(stream, retry);
+            this.stream = retryStream;
+            return retryStream;
+          }
+        } else {
+          const e = GoogleError.parseGRPCStatusDetails(error);
+          e.note =
+            'Exception occurred in retry method that was ' +
+            'not classified as transient';
+          this.emit('error', error);
+          this.destroy(error);
+          return;
+        }
+      } else {
+        return GoogleError.parseGRPCStatusDetails(error);
+      }
+    });
+    return retryStream;
+  }
+
+  resetStreams(requestStream: CancellableStream) {
+    if (requestStream) {
+      requestStream.cancel && requestStream.cancel();
+
+      if (requestStream.destroy) {
+        requestStream.destroy();
+      } else if (requestStream.end) {
+        requestStream.end();
+      }
+    }
+  }
+
   /**
    * Specifies the target stream.
    * @param {ApiCall} apiCall - the API function to be called.
    * @param {Object} argument - the argument to be passed to the apiCall.
+   * @param {RetryOptions} retry - Configures the exceptions upon which the
+   *   function eshould retry, and the parameters to the exponential backoff retry
+   *   algorithm.
    */
   setStream(
     apiCall: SimpleCallbackFunction,
     argument: {},
-    retryRequestOptions: RetryRequestOptions = {}
+    retryRequestOptions: RetryRequestOptions = {},
+    retry: RetryOptions
   ) {
+    this.apiCall = apiCall;
+    this.argument = argument;
+
     if (this.type === StreamType.SERVER_STREAMING) {
       if (this.rest) {
         const stream = apiCall(argument, this._callback) as CancellableStream;
         this.stream = stream;
         this.setReadable(stream);
+      } else if (this.new_retry) {
+        warn(
```

Comment:
Do we _really_ need to scare the customers with this?

Reply:
You tell me! I think this came from a place of wanting to be informative and as responsible as possible, but if it seems like it'd do more harm than good I can remove it!

Reply:
I think let's remove the warning. When we ship this code, it will be considered GA, not beta or something, and emitting a warning will probably send a wrong message.